consumers/hidden-test-consumer/src/hidden_test_consumer/process.py

```
@@ -0,0 +1,298 @@
+
+    def __init__(
+        self, problem_id: int, client_id: str, bucket_name: str, grpc_server: str
+    ):
+        self.logger = setup_logger("HiddenTestProcessor")
+
+        self.problem_id = problem_id
+        self.client_id = client_id
+        self.bucket_name = bucket_name
+        self.grpc_server = grpc_server
+
+        self.download_dir = os.environ.get("DOWNLOAD_DIR", "/tmp")
```

Ruff (0.8.2)

Probable insecure usage of temporary file or directory: "/tmp"

(S108)